activity.py

```diff
@@ -402,30 +403,27 @@ def guess_resource(self):
             if previous_activity.resource:
                 self.resource = previous_activity.resource
                 self.party = previous_activity.resource.party
+                if not self.party:
+                    self.party = self.on_change_with_party()
         elif self.origin and isinstance(self.origin, ElectronicMail):
-            _, email = parseaddr(self.origin.from_)
+            # parseaddr return first email
+            _, email_from = parseaddr(self.origin.from_)
+            _, email_to = parseaddr(self.origin.to)
             rejected_emails = self.emails_to_reject()
-            if email in rejected_emails:
-                addresses = [x[1] for x in self.origin.all_to
-                    if x[1] not in rejected_emails]
-                if not addresses:
-                    return
-                email = addresses[0]
-
-            activities = Activity.search([
-                    ('party', '!=', None),
-                    ('origin.from_', 'ilike', '%' + email + '%',
-                        'electronic.mail'),
-                    ], limit=1, order=[('dtstart', 'DESC')])
-            if activities:
-                self.party = activities[0].party
+            addresses = [x for x in (email_from, email_to)
+                if x not in rejected_emails and x != '']
+            if not addresses:
                 return
 
-            activities = Activity.search([
-                    ('party', '!=', None),
+            email = addresses[0]
+            activities = Activity.search(['OR', [
```

Comment:
De dues queries simplifiquem amb un 'OR' en un sol "search".

Reply:
T'estàs carregant el ('party', '!=', None) però l'has de mantenir!

Reply:
I si no està informat del tercer, i en canvi trobes el origin.from_ o origin.to per relacionar-lo? Si encara no has creat el tercer, "no trobaràs certs registres" al afegir aquesta condició.

Reply:
Si no té tercer no m'interessa perquè el que fem a la línia 428 és precisament assignar el tercer de l'activitat, així que si no té tercer, no l'hem d'agafar.